core/src/main/java/io/questdb/griffin/engine/ops/CopyExportFactory.java

```diff
@@ -0,0 +1,305 @@
+import io.questdb.cairo.sql.RecordCursorFactory;
+import io.questdb.cairo.sql.TableMetadata;
+import io.questdb.cutlass.parquet.CopyExportRequestTask;
+import io.questdb.cutlass.text.CopyExportContext;
+import io.questdb.cutlass.text.CopyExportResult;
+import io.questdb.cutlass.text.CopyImportContext;
+import io.questdb.griffin.CompiledQuery;
+import io.questdb.griffin.PlanSink;
+import io.questdb.griffin.SqlCompiler;
+import io.questdb.griffin.SqlException;
+import io.questdb.griffin.SqlExecutionContext;
+import io.questdb.griffin.engine.SingleValueRecordCursor;
+import io.questdb.griffin.model.CopyModel;
+import io.questdb.griffin.model.ExpressionNode;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.mp.RingQueue;
+import io.questdb.mp.SPSequence;
+import io.questdb.network.SuspendEvent;
+import io.questdb.std.GenericLexer;
+import io.questdb.std.Misc;
+import io.questdb.std.Numbers;
+import io.questdb.std.str.StringSink;
+import org.jetbrains.annotations.Nullable;
+
+import static io.questdb.std.GenericLexer.unquote;
+
+/**
+ * Executes COPY statement lazily, i.e. on record cursor initialization, to play
+ * nicely with server-side statements in PG Wire and query caching in general.
+ */
+public class CopyExportFactory extends AbstractRecordCursorFactory {
+
+    private static final Log LOG = LogFactory.getLog(CopyExportFactory.class);
+    private final static GenericRecordMetadata METADATA = new GenericRecordMetadata();
+    private final StringSink exportIdSink = new StringSink();
+    private final CopyImportFactory.CopyRecord record = new CopyImportFactory.CopyRecord();
+    private final SingleValueRecordCursor cursor = new SingleValueRecordCursor(record);
+    private int compressionCodec;
+    private int compressionLevel;
+    private CopyExportContext copyContext;
+    private CopyExportResult copyExportResult;
+    private int dataPageSize;
+    private String fileName;
+    private MessageBus messageBus;
+    private int parquetVersion;
+    private int partitionBy;
+    private boolean rawArrayEncoding = false;
+    private int rowGroupSize;
+    private SecurityContext securityContext;
+    private String selectText = null;
+    private int sizeLimit;
+    private CharSequence sqlText;
+    private boolean statisticsEnabled;
+    private @Nullable SuspendEvent suspendEvent = null;
+    private @Nullable String tableName = null;
+    private int tableOrSelectTextPos = 0;
+    private boolean userSpecifiedExportOptions;
+
+    public CopyExportFactory(
+            MessageBus messageBus,
+            CopyExportContext copyContext,
+            CopyModel model,
+            SecurityContext securityContext,
+            CharSequence sqlText
+    ) throws SqlException {
+        super(METADATA);
+        this.of(messageBus, copyContext, model, null, securityContext, sqlText);
+    }
+
+    public CopyExportFactory(
+            MessageBus messageBus,
+            CopyExportContext copyContext,
```

Comment:
parameter names must not obscure the purpose of the argument. They have to be consistent with args passed in. If input it "exportContext" it must to degrade to ambiguous "context"